src/Autofac.Integration.Mef/RegistrationExtensions.cs

```diff
@@ -423,10 +428,126 @@ private static void ProcessExportDefinition(ContainerBuilder builder, Func<Expor
             }
         }
 
+        private static bool TryGetLazyType(this ContractBasedImportDefinition definition, out Type resultType, out Type lazyType)
+        {
+            // There are a couple of classes that are internal that provide us some information we can use to
+            // properly gauge if we need to do our lazy activation or not.
+            var definitionType = definition.GetType();
+            LazyMemberInfo? lazyMemberInfo = null;
+            resultType = null;
+            lazyType = null;
+            switch (definitionType.Name)
+            {
+                // The first case is the base class of the second case for both of these pairs.
+                case "ReflectionMemberImportDefinition":
+                case "PartCreatorMemberImportDefinition":
+                    lazyMemberInfo = ReflectionModelServices.GetImportingMember(definition);
+                    break;
+                case "ReflectionParameterImportDefinition":
+                case "PartCreatorParameterImportDefinition":
+                    resultType = ReflectionModelServices.GetImportingParameter(definition)?.Value?.ParameterType;
+                    break;
+                default:
+                    return false;
+            }
+
+            if (lazyMemberInfo.HasValue)
+            {
+                foreach (var accessor in lazyMemberInfo.Value.GetAccessors())
+                {
+                    if (accessor is MethodInfo methodInfo)
+                    {
+                        // This is either a getter or a setter.
+                        resultType = methodInfo.ReturnType;
+                        if (resultType == typeof(void))
+                        {
+                            resultType = methodInfo.GetParameters()[0].ParameterType;
+                        }
+
+                        break;
+                    }
+                    else if (accessor is FieldInfo fieldInfo)
+                    {
+                        resultType = fieldInfo.FieldType;
+                        break;
+                    }
+                }
+            }
+
+            if (resultType != null && resultType != typeof(void))
+            {
+                // Have to handle 2 cases
+                // Single cardinality = Lazy<T, TMetadata>
+                // Multiple cardinality = IEnumerable<Lazy<T, TMetadata>>
+                bool isLazy;
+                if (
+                    (isLazy = resultType.IsGenericTypeDefinedBy(typeof(Lazy<,>)))
+                        ||
+                    (
+                    resultType.IsGenericTypeDefinedBy(typeof(IEnumerable<>))
+                        &&
+                    resultType.GetGenericArguments()[0].IsGenericTypeDefinedBy(typeof(Lazy<,>))))
```

Comment:
```suggestion
                if (
                     (isLazy = resultType.IsGenericTypeDefinedBy(typeof(Lazy<,>)))
                       ||
                     (
                       resultType.IsGenericTypeDefinedBy(typeof(IEnumerable<>))
                         &&
                       resultType.GetGenericArguments()[0].IsGenericTypeDefinedBy(typeof(Lazy<,>))))
```

This is more about my OCD than your code. 😆

Reply:
Truthfully, I wasn't expecting this PR to be accepted after minor revisions so I turned my OCD way down.